app/models/manageiq/providers/openstack/infra_manager/host/operations.rb

```diff
@@ -0,0 +1,40 @@
+module ManageIQ::Providers::Openstack::InfraManager::Host::Operations
+  def self.included(base)
+    base.send :include, InstanceMethods
+    base.extend ClassMethods
+  end
+
+  module InstanceMethods
+    def nova_system_service
+      # we need to be sure that host has compute service
+      system_services.find_by(:name => 'openstack-nova-compute')
+    end
+
+    def nova_fog_service
+      # TODO: check if host is part of OpenStack Infra
+      # host's cluster needs cloud assigned
+      cloud = ems_cluster.cloud
+      # hostname of host in hypervisor is used to properly select service from OpenStack
+      host_name = hypervisor_hostname
+      fog_services = cloud.openstack_handle.compute_service.services
+      fog_services.find { |s| s.host =~ /#{host_name}/ && s.binary == 'nova-compute' }
+    end
+
+    def nova_fog_enable_service
+      nova_fog_service.enable
+    end
+
+    def nova_fog_disable_service
```

Comment:
Yeah, I prefer to not prefix the methods with `nova_fog`.

Also, is `'nova-compute'` going to be the only service that gets disabled/enabled?